FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m

```diff
@@ -1164,6 +1207,95 @@ - (void)testConfigureConfigWithValidInput {
   XCTAssertEqual([config valueForKey:@"_appName"], kFIRDefaultAppName);
 }
 
+#pragma mark app extension.
+- (void)testETagAndFetchTimeUpdatedInSharedUserDefaultsForAppExtension {
+  NSMutableArray<XCTestExpectation *> *expectations =
+      [[NSMutableArray alloc] initWithCapacity:RCNTestRCNumTotalInstances];
+  for (int i = 0; i < 1; i++) {
+    expectations[i] = [self
+        expectationWithDescription:
+            [NSString stringWithFormat:@"Test fetch configs for app extension - instance %d", i]];
+    XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusNoFetchYet);
+    FIRRemoteConfigFetchCompletion fetchCompletion = ^void(FIRRemoteConfigFetchStatus status,
+                                                           NSError *error) {
+      XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusSuccess);
+      XCTAssertNil(error);
+      XCTAssertNotNil(_configInstances[i].lastFetchTime);
+      XCTAssertGreaterThan(_configInstances[i].lastFetchTime.timeIntervalSince1970, 0,
+                           @"last fetch time interval should be set.");
+      NSUserDefaults *externalUserDefaults =
+          [[NSUserDefaults alloc] initWithSuiteName:[self firstAppOptions].appGroupID];
+      NSString *currentETag = [externalUserDefaults objectForKey:@"frc.latestETag"];
+      NSNumber *lastFetchTime = [externalUserDefaults objectForKey:@"frc.lastSuccessfulFetchTime"];
+      XCTAssertTrue([currentETag isEqualToString:@"etag1-0"]);
+      NSDate *lastFetchedDateTime = _configInstances[i].lastFetchTime;
+      XCTAssertEqual(lastFetchTime.doubleValue, [lastFetchedDateTime timeIntervalSince1970]);
+      // Update the eTag
+      XCTAssertTrue([currentETag isEqualToString:@"etag1-0"]);
+      [expectations[i] fulfill];
+    };
+    [_configInstances[i] fetchWithExpirationDuration:43200 completionHandler:fetchCompletion];
+  }
+
+  [self waitForExpectationsWithTimeout:_expectationTimeout
+                               handler:^(NSError *error) {
+                                 XCTAssertNil(error);
+                               }];
+}
+
+// An app extension should be allowed to fetch within the minimumFetchInterval if the main app/other
+// extensions have fetched updated config since.
+- (void)testFetchConfigsForAppExtension {
```

Comment:
This test is pretty hard to follow. As mentioned above, more comments and some blank lines to separate unrelated blocks of code would be very helpful.

Reply:
Added more comments to test and made test code more readable.